src/components/chip-set/chip.types.ts

```diff
@@ -19,6 +20,11 @@ export interface Chip<T = any> {
      */
     icon?: string | Icon;
 
+    /**
+     * A picture to be displayed instead of the icon on the chip.
+     */
+    image?: Img;
+
```

Comment:
I never quite understood why we are repeating some of these props in `limel-chip` again. But just like `icon`, I repeated this too.

Reply:
I'm not sure I understand what repetition you are referring to. Are you talking about defining `image?: Img;` here and defining `@Prop() public image?: Img;` in `src/components/chip/chip.tsx`? Or something else? 😊

Reply:
yes exactly.

Reply:
That's because the definition here defines what the "data object" representing a chip should look like, while the definition in `src/components/chip/chip.tsx` defines what properties the component _rendering_ a chip accepts.

It's no coincidence that there's an overlap there… 😄

Reply:
I still don't understand why we just didn't `extend` the Chip types from chipset, inside `limel-chip`.

We do similar things for instance here:

https://github.com/Lundalogik/lime-elements/blob/7fe1fbbd5af68cb064118af75e3c6985b881831e/src/components/progress-flow/progress-flow.types.ts#L7

Reply:
If I understand you correctly, you want to do something like this inside the component?
```ts
export class Chip extends ChipInterface
```

That will not work since the component is a class and the chip type is an interface. Classes can only extend other classes, and instead need to implement an interface or a type. We could in theory create a class for the chip as well, but Stencil does not allow components to extend other classes and we would still have to repeat all the properties so we could decorate them with `@Prop`